app/javascript/sidebar-widget/SidebarWidget.jsx

```diff
@@ -1,34 +1,18 @@
-import { h, Component } from 'preact';
+import { h, Fragment } from 'preact';
+import { useCallback, useEffect, useState } from 'preact/hooks';
 import sendFollowUser from '../utilities/sendFollowUser';
 import SidebarUser from './sidebarUser';
 
-class SidebarWidget extends Component {
-  constructor(props) {
-    super(props);
-    this.getSuggestedUsers = this.getSuggestedUsers.bind(this);
-    this.getTagInfo = this.getTagInfo.bind(this);
-    this.followUser = this.followUser.bind(this);
-    this.state = {
-      tagInfo: {},
-      suggestedUsers: [],
-    };
-  }
-
-  componentDidMount() {
-    this.getTagInfo();
-    this.getSuggestedUsers();
-  }
+const SidebarWidget = () => {
+  const [suggestedUsers, setSuggestedUsers] = useState([]);
 
-  getTagInfo() {
-    this.setState({
-      tagInfo: JSON.parse(
+  useEffect(() => {
+    const tagInfo =
+      JSON.parse(
         document.getElementById('sidebarWidget__pack').dataset.tagInfo,
-      ),
-    });
-  }
+      ) || {};
```

Comment:
Refactor (blocking): Setting a default value here does not seem necessary. We are always expecting `tagInfo` to be an object. If it isn't, Honeybader catches the unexpected error and reports it.

If we give it a a default value of `{}`, a valid request will be made that returns an empty array for the tag `undefined`.